sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -37,8 +37,18 @@ import org.apache.spark.sql.types._
  */
 private[sql] class JacksonGenerator(
```

Comment:
`JacksonGenerator` isn't an API so it doesn't make much sense to make it pluggable

Reply:
As far I can see datasources are sometimes implemented under `org.apache.spark.sql`, thus accessing private and protected members, ie. https://github.com/datastax/spark-cassandra-connector/tree/master/connector/src/main/scala/org/apache/spark/sql/cassandra
Of course relying on private API could lead to unpredictable breaking changes. Ideally `JacksonParser` and `JacksonGenerator` could evolve to a public API usable by any binary dataformat implementation.